osbuild/main_cli.py

```diff
@@ -97,6 +98,9 @@ def osbuild_cli():
 
     index = osbuild.meta.Index(args.libdir)
 
+    # initialize error ids
+    OSbuildError.add_error_id(SourceException)
```

Comment:
Global dynamic error ids kinda defies the point of having format versioned error reporting. The whole idea behind our formatting abstraction is to separate the internal structures and their outside representation. The same should be true for our error reporting, i.e. we just work with `SourceError`, `BuildError`, ... internally and then represent those in a format specific way to the outside.

Reply:
I agree to the separation. However, I think we need a way to "declare" the errors. I mean declaring in a way that we have a list of "This error means this number".

Because on Composer's side, we need to rely on the ID to reconstruct the meaning of the error. I figured it would be easier for a Composer dev to have the list accessible in the main.

Reply:
However, I'm making an experiment in the code, you tell me if it's more of your liking ;)

I statically say that a Unknown SourceError is of value 1000. Then, when I want to generate an ID, I use the base value + the kind of error.

The advantage being that we can have different numbers depending on what is generated by the subcomand.